### ISSUE TITLE
Deploying Custom CNN implementations (fasterRCNN, maskRCNN etc) using TF-Serving

### ISSUE THREAD

**Issue by rishabhmalhotra** [tensorflow/tensorflow#17519]
Running popular forks (fasterRCNN, maskRCNN), one finds they cannot be serialized as a GraphDef because of the usage of py_funcs. (Ref: Official tf documentation)

The error observed is such: 
``2018-03-07 21:26:02.451945: E tensorflow_serving/util/retrier.cc:38] Loading servable: {name: saved_model version: 1} failed: Not found: Op type not registered 'PyFunc' in binary running on f39ddd6ef2c7. Make sure the Op and Kernel are registered in the binary running in this process``

Is there any official fork/implementation for these popular frameworks or is there some documentation which can be used to convert py_funcs to native tensorflow operations?

Also, kindly let me know if any of the new tf betas in the works support/will support it since I am looking to deploy mine.

Thanks

-----------------------------------------------
System Info:
  - Running in docker container (ubuntu/bash)
  - tf 1.5.x

**Comment by rishabhmalhotra** [comment]
@chrisolston any input(s) would be appreciated, thanks!

**Comment by Cristiano-3** [comment]
I got the same problem, while serving text-detection-cptn:
2018-03-12 15:21:45.537829: E tensorflow_serving/core/aspired_versions_manager.cc:352] Servable {name: ctpn version: 1} cannot be loaded: Not found: Op type not registered 'PyFunc' in binary running on skyriver. Make sure the Op and Kernel are registered in the binary running in this process.

**Comment by rishabhmalhotra** [comment]
@Cristiano-3 yes, this seems to be a limitation of Tensorflow Serving as of now. It would really make sense to deploy production scale ML code with tf-serving once either this is resolved and support is added or a workaround is found for the same since most of prod code I've worked uses some sort of custom framework(it's almost never as simple as deploying a super simple neural net without custom frameworks).

I understand that this issue arises due to the fact that Tf is a library written for python in C++.
Would love to get some direction and/or contribute to this process of adding pyfunc support to tensorflow serving.

**Comment by drpngx** [comment]
@sukritiramesh any comment here?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @drpngx: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by drpngx** [comment]
@nfiedel any thoughts?

**Comment by drpngx** [comment]
In that case, actually I would recommend implementing `sinh` using tensorflow based on the [formula](https://en.wikipedia.org/wiki/Hyperbolic_function).

**Comment by rishabhmalhotra** [comment]
Thanks for the input but I'm still looking to resolve the core issue, which is to actually get py_funcs working with tensorflow serving. This is majorly due to the vast resources available in terms of custom framework implementations like FasterRCNN.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @drpngx: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @drpngx: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by drpngx** [comment]
So I think it's just a matter of linking the op in to your custom binary. Did you try that?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @drpngx: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by rishabhmalhotra** [comment]
Yes that does seem to work, thanks for your time @drpngx.
